<img src=".\images\Logo.png" width=150 align="left" /> <img src=".\images\Logo2.jpg" width=450 align="right" />


# <center><font color= #1e8449 > <b>CU18_Infraestructuras_eventos</font></center>


<font color='steelblue'><b>Citizenlab Data Science Methodology > II - Data Processing Domain </font>
***
> # <font color='steelblue'> <b>05.- Data Collection</font>

<font color='steelblue'>Data Collection is the process to obtain and generate (if required) necessary data to model the problem.</font>

### 18. Agrupar estaciones de metro

* Los datos de metro están geolocalizados por "boca" de metro
* La capacidad es previsible que esté por estación (o como mucho estación-línea)
* Así que hay que agrupar, en esta tarea se hace por estación.








<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Settings" data-toc-modified-id="Settings-1"><font color="green">Settings</font></a></span></li><li><span><a href="#Data-Load" data-toc-modified-id="Data-Load-2"><font color="green">Data Load</font></a></span></li><li><span><a href="#ETL-Processes" data-toc-modified-id="ETL-Processes-3">ETL Processes</a></span><ul class="toc-item"><li><span><a href="#Import-data-from:-CSV,-Excel,-Tab,-JSON,-SQL,-and-Parquet-files" data-toc-modified-id="Import-data-from:-CSV,-Excel,-Tab,-JSON,-SQL,-and-Parquet-files-3.1">Import data from: CSV, Excel, Tab, JSON, SQL, and Parquet files</a></span></li></ul></li><li><span><a href="#Synthetic-Data-Generation" data-toc-modified-id="Synthetic-Data-Generation-4">Synthetic Data Generation</a></span></li><li><span><a href="#Fake-Data-Generation" data-toc-modified-id="Fake-Data-Generation-5">Fake Data Generation</a></span></li><li><span><a href="#Open-Data" data-toc-modified-id="Open-Data-6">Open Data</a></span></li><li><span><a href="#Data-Save" data-toc-modified-id="Data-Save-7"><font color="green">Data Save</font></a></span></li><li><span><a href="#Main-Conclusions" data-toc-modified-id="Main-Conclusions-8"><font color=" #2874a6 ">Main Conclusions</font></a></span></li><li><span><a href="#Main-Actions" data-toc-modified-id="Main-Actions-9"><font color=" #2874a6 ">Main Actions</font></a></span><ul class="toc-item"><li><ul class="toc-item"><li><span><a href="#Acciones-done" data-toc-modified-id="Acciones-done-9.0.1"><font color=" #2874a6 ">Acciones done</font></a></span></li><li><span><a href="#Acctions-to-perform" data-toc-modified-id="Acctions-to-perform-9.0.2"><font color=" #2874a6 ">Acctions to perform</font></a></span></li></ul></li></ul></li></ul></div>

## <font color='green'>Settings</font>

### Packages to use

_ELIMINAR O AÑADIR LO QUE TOQUE. COPIAR VERSIONES AL FINAL Y QUITAR CÓDIGO DE VERSIONES_

- {tcltk} para selección interactiva de archivos locales
- {sf} para trabajar con georeferenciación
- {readr} para leer y escribir archivos csv
- {dplyr} para explorar datos
- {stringr} para manipulación de cadenas de caracteres
- {tidyr} para organización de datos


In [1]:
library(sf)
library(readr)
library(dplyr)
# library(stringr)
library(tidyr)

p <- c("tcltk", "sf", "readr", "dplyr", "stringr", "tidyr")


Linking to GEOS 3.10.2, GDAL 3.4.2, PROJ 8.2.1; sf_use_s2() is TRUE


Attaching package: 'dplyr'


The following objects are masked from 'package:stats':

    filter, lag


The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union




### Paths

In [2]:
iPath <- "Data/Input/"
oPath <- "Data/Output/"

## <font color='green'>Data Load</font>

If there are more than one input file, make as many sections as files to import.

Instrucciones
- Los ficheros de entrada del proceso están siempre en Data/Input/.   
- Si hay más de un fichero de entrada, se crean tantos objetos iFile_xx y file_data_xx como ficheros de entrada (xx número correlativo con dos dígitos, rellenar con ceros a la izquierda)

<font color='tomato'><b> OPCION A:</b> Seleccionar fichero en ventana para mayor comodidad</font> 

Data load using the {tcltk} package. Ucomment the line if not using this option

In [3]:
# file_data <- tcltk::tk_choose.files(multi = FALSE)

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>



In [4]:
iFile <- "CU_18_05_17_infraestructuras.csv"
file_data <- paste0(iPath, iFile)

if(file.exists(file_data)){
    cat("Se leerán datos del archivo: ", file_data)
} else{
    warning("Cuidado: el archivo no existe.")
}


Se leer<U+00E1>n datos del archivo:  Data/Input/CU_18_05_17_infraestructuras.csv

#### Data file to dataframe

Usar la función adecuada según el formato de entrada (xlsx, csv, json, ...)

In [5]:
data <- read_csv(file_data)


Rows: 1633 Columns: 18
-- Column specification --------------------------------------------------------
Delimiter: ","
chr  (8): grupo, tipo, nombre, CODMUN, DIRECCION, info, CMUN, CDIS
dbl (10): dist_aeropuerto, dist_intercambiador, dist_cercanias, dist_hospita...

i Use `spec()` to retrieve the full column specification for this data.
i Specify the column types or set `show_col_types = FALSE` to quiet this message.


Estructura de  los datos:

In [6]:
glimpse(data)

Rows: 1,633
Columns: 18
$ grupo               <chr> "Transporte", "Transporte", "Transporte", "Transpo~
$ tipo                <chr> "Intercambiadores", "Intercambiadores", "Intercamb~
$ nombre              <chr> "Grandes Intercambiadores Plaza El<U+00ED>ptica", "Grande~
$ CODMUN              <chr> "079", "079", "079", "079", "079", "079", "079", "~
$ DIRECCION           <chr> "Plaza El<U+00ED>ptica s/n", "Calle Princesa, 89", "Estac~
$ info                <chr> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA~
$ CMUN                <chr> "079", "079", "079", "079", "079", "079", "079", "~
$ CDIS                <chr> "12", "09", "09", "05", "05", "05", "06", "02", "0~
$ dist_aeropuerto     <dbl> 5.2031430, 8.5507475, 7.3002463, 9.9339069, 9.1988~
$ dist_intercambiador <dbl> 0.00000000, 0.00000000, 0.00000000, 0.00000000, 0.~
$ dist_cercanias      <dbl> 1.648973148, 1.468638183, 0.088338692, 0.736877403~
$ dist_hospital       <dbl> 1.8154156, 0.4221115, 1.3855767, 1.0733484, 0.3488~
$ 

Muestra de datos:

In [7]:
slice_head(data, n = 5)

grupo      tipo            
1 Transporte Intercambiadores
2 Transporte Intercambiadores
3 Transporte Intercambiadores
4 Transporte Intercambiadores
5 Transporte Intercambiadores
  nombre                                              CODMUN
1 Grandes Intercambiadores Plaza El<U+00ED>ptica      079   
2 Grandes Intercambiadores Moncloa                    079   
3 Grandes Intercambiadores Pr<U+00ED>ncipe P<U+00ED>o 079   
4 Grandes Intercambiadores Plaza de Castilla          079   
5 Grandes Intercambiadores Avenida de Am<U+00E9>rica  079   
  DIRECCION                                  info CMUN CDIS dist_aeropuerto
1 Plaza El<U+00ED>ptica s/n                  NA   079  12   5.203143       
2 Calle Princesa, 89                         NA   079  09   8.550748       
3 Estaci<U+00F3>n Pr<U+00ED>ncipe P<U+00ED>o NA   079  09   7.300246       
4 Calle San Aquilino y Avenida de Asturias   NA   079  05   9.933907       
5 Avenida de Am<U+00E9>rica 9 A              NA   079  05   9.198897       
  dist_intercambiador dist_cercanias dist_hospital dist_universidad dist_gob 
1 0                   1.64897315     1.8154156     1.29960755       0.6177235
2 0                   1.46863818     0.4221115     0.30460781       0.6263847
3 0                   0.08833869     1.3855767     0.92076145       1.1700512
4 0                   0.73687740     1.0733484     0.08835165       0.1490496
5 0                   1.62310148     0.3488060     0.50695045       1.1979177
  dist_embajada dist_ccomercial X         Y       
1 0.39213238    0.7311571       -3.716577 40.38540
2 1.45739694    1.6105674       -3.719508 40.43474
3 5.62455975    0.9060857       -3.719560 40.42080
4 0.08316773    0.7758344       -3.689044 40.46719
5 0.40303107    0.9512580       -3.676731 40.43797

## ETL Processes

### Import data from: CSV, Excel, Tab, JSON, SQL, and Parquet files

Se han importado en el apartado Data Load anterior:

* Infraestructuras completas con distancias

Incluir apartados si procede para: Extracción de datos (select, filter), Transformación de datos, (mutate, joins, ...). Si es necesario tratar datos perdidos, indicarlo también en NB 09.2

## Data transform

* Conversión de coordenadas numéricas a objeto espacial

In [8]:
tdata_sf <- data |> 
    st_as_sf(coords = c("X", "Y"))

In [9]:
tdata_sf |> glimpse()

Rows: 1,633
Columns: 17
$ grupo               <chr> "Transporte", "Transporte", "Transporte", "Transpo~
$ tipo                <chr> "Intercambiadores", "Intercambiadores", "Intercamb~
$ nombre              <chr> "Grandes Intercambiadores Plaza El<U+00ED>ptica", "Grande~
$ CODMUN              <chr> "079", "079", "079", "079", "079", "079", "079", "~
$ DIRECCION           <chr> "Plaza El<U+00ED>ptica s/n", "Calle Princesa, 89", "Estac~
$ info                <chr> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA~
$ CMUN                <chr> "079", "079", "079", "079", "079", "079", "079", "~
$ CDIS                <chr> "12", "09", "09", "05", "05", "05", "06", "02", "0~
$ dist_aeropuerto     <dbl> 5.2031430, 8.5507475, 7.3002463, 9.9339069, 9.1988~
$ dist_intercambiador <dbl> 0.00000000, 0.00000000, 0.00000000, 0.00000000, 0.~
$ dist_cercanias      <dbl> 1.648973148, 1.468638183, 0.088338692, 0.736877403~
$ dist_hospital       <dbl> 1.8154156, 0.4221115, 1.3855767, 1.0733484, 0.3488~
$ 

In [10]:
tdata_sf |> slice_head(n = 5) |> tibble()

grupo      tipo            
1 Transporte Intercambiadores
2 Transporte Intercambiadores
3 Transporte Intercambiadores
4 Transporte Intercambiadores
5 Transporte Intercambiadores
  nombre                                              CODMUN
1 Grandes Intercambiadores Plaza El<U+00ED>ptica      079   
2 Grandes Intercambiadores Moncloa                    079   
3 Grandes Intercambiadores Pr<U+00ED>ncipe P<U+00ED>o 079   
4 Grandes Intercambiadores Plaza de Castilla          079   
5 Grandes Intercambiadores Avenida de Am<U+00E9>rica  079   
  DIRECCION                                  info CMUN CDIS dist_aeropuerto
1 Plaza El<U+00ED>ptica s/n                  NA   079  12   5.203143       
2 Calle Princesa, 89                         NA   079  09   8.550748       
3 Estaci<U+00F3>n Pr<U+00ED>ncipe P<U+00ED>o NA   079  09   7.300246       
4 Calle San Aquilino y Avenida de Asturias   NA   079  05   9.933907       
5 Avenida de Am<U+00E9>rica 9 A              NA   079  05   9.198897       
  dist_intercambiador dist_cercanias dist_hospital dist_universidad dist_gob 
1 0                   1.64897315     1.8154156     1.29960755       0.6177235
2 0                   1.46863818     0.4221115     0.30460781       0.6263847
3 0                   0.08833869     1.3855767     0.92076145       1.1700512
4 0                   0.73687740     1.0733484     0.08835165       0.1490496
5 0                   1.62310148     0.3488060     0.50695045       1.1979177
  dist_embajada dist_ccomercial geometry                  
1 0.39213238    0.7311571       POINT (-3.716577 40.3854) 
2 1.45739694    1.6105674       POINT (-3.719508 40.43474)
3 5.62455975    0.9060857       POINT (-3.71956 40.4208)  
4 0.08316773    0.7758344       POINT (-3.689044 40.46719)
5 0.40303107    0.9512580       POINT (-3.676731 40.43797)

#### Data extract

* Extraer solo tipo bocas de metro

In [11]:
edata <- tdata_sf |>
    filter(tipo == "Bocas de metro")

In [12]:
edata |> glimpse()

Rows: 771
Columns: 17
$ grupo               <chr> "Transporte", "Transporte", "Transporte", "Transpo~
$ tipo                <chr> "Bocas de metro", "Bocas de metro", "Bocas de metr~
$ nombre              <chr> "Plaza de Castilla - Castellana", "Plaza de Castil~
$ CODMUN              <chr> "079", "079", "079", "079", "079", "079", "079", "~
$ DIRECCION           <chr> "Frente dep<U+00F3>sito C. Isabel II", "Paseo de la Caste~
$ info                <chr> "L<U+00ED>neas: 1, 10, 9", "L<U+00ED>neas: 1, 10, 9", "L<U+00ED>neas: 1~
$ CMUN                <chr> "079", "079", "079", "079", "079", "079", "079", "~
$ CDIS                <chr> "05", "06", "06", "06", "06", "06", "06", "06", "0~
$ dist_aeropuerto     <dbl> 9.905774, 9.942879, 9.974996, 10.482757, 10.338293~
$ dist_intercambiador <dbl> 0.190340745, 0.041478689, 0.041290988, 0.666464368~
$ dist_cercanias      <dbl> 0.8811005, 0.7749713, 0.7651209, 1.3860106, 1.2245~
$ dist_hospital       <dbl> 1.0921155, 1.0997205, 1.0512395, 1.2298147

#### Data transform

* Obtener estación de metro
    - En el nombre, está estación - salida

In [13]:
tdata_estacion <- edata |>
  mutate(tipo = "Estaciones de metro") |> 
  separate(nombre, c("nombre", "salida"), sep = " - ")

Warning message:
"Expected 2 pieces. Additional pieces discarded in 41 rows [75, 82, 133, 159,
160, 189, 190, 304, 305, 312, 313, 359, 367, 368, 370, 371, 372, 373, 374, 375,
...]."


In [14]:
tdata_estacion |> glimpse()

Rows: 771
Columns: 18
$ grupo               <chr> "Transporte", "Transporte", "Transporte", "Transpo~
$ tipo                <chr> "Estaciones de metro", "Estaciones de metro", "Est~
$ nombre              <chr> "Plaza de Castilla", "Plaza de Castilla", "Plaza d~
$ salida              <chr> "Castellana", "Plaza de Castilla", "P<U+00BA> Castellana,~
$ CODMUN              <chr> "079", "079", "079", "079", "079", "079", "079", "~
$ DIRECCION           <chr> "Frente dep<U+00F3>sito C. Isabel II", "Paseo de la Caste~
$ info                <chr> "L<U+00ED>neas: 1, 10, 9", "L<U+00ED>neas: 1, 10, 9", "L<U+00ED>neas: 1~
$ CMUN                <chr> "079", "079", "079", "079", "079", "079", "079", "~
$ CDIS                <chr> "05", "06", "06", "06", "06", "06", "06", "06", "0~
$ dist_aeropuerto     <dbl> 9.905774, 9.942879, 9.974996, 10.482757, 10.338293~
$ dist_intercambiador <dbl> 0.190340745, 0.041478689, 0.041290988, 0.666464368~
$ dist_cercanias      <dbl> 0.8811005, 0.7749713, 0.7651209, 1.

* Agrupar datos de bocas de metro en su estación
    - Las distancias nos quedamos con la mínima de todas las bocas
    - El resto, el primer valor (se supone que serían todos iguales)
    - La geometría, el centroide de todas las bocas

In [15]:
tdata_agrupado <- tdata_estacion |> 
  mutate(tipo = "Estaciones de metro") |> 
  separate(nombre, c("nombre", "salida"), sep = " - ") |> 
  group_by(grupo, tipo, nombre) |>
  summarise(across(CODMUN:CDIS, first),
            across(dist_aeropuerto:dist_ccomercial, min)) |> 
  st_centroid()


Warning message:
"Expected 2 pieces. Missing pieces filled with `NA` in 771 rows [1, 2, 3, 4, 5,
6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...]."
`summarise()` has grouped output by 'grupo', 'tipo'. You can override using the
`.groups` argument.
Warning message in st_centroid.sf(summarise(group_by(separate(mutate(tdata_estacion, :
"st_centroid assumes attributes are constant over geometries of x"


* Geometrías a coordenadas numéricas para exportar a csv

In [16]:
tdata_metro <- tdata_agrupado |>
    bind_cols(st_coordinates(tdata_agrupado)) |>
    st_drop_geometry()

* Sustituir datos de metro 

In [17]:
tdata_out <-
    data |>
    filter(tipo != "Bocas de metro") |>
    bind_rows(tdata_metro)

In [18]:
tdata_out |> glimpse()

Rows: 1,138
Columns: 18
$ grupo               <chr> "Transporte", "Transporte", "Transporte", "Transpo~
$ tipo                <chr> "Intercambiadores", "Intercambiadores", "Intercamb~
$ nombre              <chr> "Grandes Intercambiadores Plaza El<U+00ED>ptica", "Grande~
$ CODMUN              <chr> "079", "079", "079", "079", "079", "079", "079", "~
$ DIRECCION           <chr> "Plaza El<U+00ED>ptica s/n", "Calle Princesa, 89", "Estac~
$ info                <chr> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA~
$ CMUN                <chr> "079", "079", "079", "079", "079", "079", "079", "~
$ CDIS                <chr> "12", "09", "09", "05", "05", "05", "06", "02", "0~
$ dist_aeropuerto     <dbl> 5.2031430, 8.5507475, 7.3002463, 9.9339069, 9.1988~
$ dist_intercambiador <dbl> 0.0000000, 0.0000000, 0.0000000, 0.0000000, 0.0000~
$ dist_cercanias      <dbl> 1.648973148, 1.468638183, 0.088338692, 0.736877403~
$ dist_hospital       <dbl> 1.8154156, 0.4221115, 1.3855767, 1.0733484, 0.3488~
$ 

In [19]:
tdata_out |> slice_head(n = 5)

grupo      tipo            
1 Transporte Intercambiadores
2 Transporte Intercambiadores
3 Transporte Intercambiadores
4 Transporte Intercambiadores
5 Transporte Intercambiadores
  nombre                                              CODMUN
1 Grandes Intercambiadores Plaza El<U+00ED>ptica      079   
2 Grandes Intercambiadores Moncloa                    079   
3 Grandes Intercambiadores Pr<U+00ED>ncipe P<U+00ED>o 079   
4 Grandes Intercambiadores Plaza de Castilla          079   
5 Grandes Intercambiadores Avenida de Am<U+00E9>rica  079   
  DIRECCION                                  info CMUN CDIS dist_aeropuerto
1 Plaza El<U+00ED>ptica s/n                  NA   079  12   5.203143       
2 Calle Princesa, 89                         NA   079  09   8.550748       
3 Estaci<U+00F3>n Pr<U+00ED>ncipe P<U+00ED>o NA   079  09   7.300246       
4 Calle San Aquilino y Avenida de Asturias   NA   079  05   9.933907       
5 Avenida de Am<U+00E9>rica 9 A              NA   079  05   9.198897       
  dist_intercambiador dist_cercanias dist_hospital dist_universidad dist_gob 
1 0                   1.64897315     1.8154156     1.29960755       0.6177235
2 0                   1.46863818     0.4221115     0.30460781       0.6263847
3 0                   0.08833869     1.3855767     0.92076145       1.1700512
4 0                   0.73687740     1.0733484     0.08835165       0.1490496
5 0                   1.62310148     0.3488060     0.50695045       1.1979177
  dist_embajada dist_ccomercial X         Y       
1 0.39213238    0.7311571       -3.716577 40.38540
2 1.45739694    1.6105674       -3.719508 40.43474
3 5.62455975    0.9060857       -3.719560 40.42080
4 0.08316773    0.7758344       -3.689044 40.46719
5 0.40303107    0.9512580       -3.676731 40.43797

Si no aplica: Estos datos no requieren tareas de este tipo.

## Synthetic Data Generation

No aplica

## Fake Data Generation

No aplica

## Open Data

Los datos originales fueron descargados de fuentes abiertas

## <font color='green'>Data Save</font>

Este proceso, puede copiarse y repetirse en aquellas partes del notebbok que necesiten guardar datos.
Recuerde cambiar las cadenas añadida del fichero para diferenciarlas

<font color='tomato'> Identificamos los datos a guardar</font>

In [20]:
data_to_save <- tdata_out


<font color='tomato'>Estructura de nombre de archivos:</font>

* Código del caso de uso, por ejemplo "CU_04"
* Número del proceso que lo genera, por ejemplo "_05".
* Número de la tarea que lo genera, por ejemplo "_01"
* En caso de generarse varios ficheros en la misma tarea, llevarán _01 _02 ... después
* Nombre: identificativo de "properData", por ejemplo "_zonasgeo"
* Extensión del archivo

Ejemplo: "CU_04_05_01_01_zonasgeo.json, primer fichero que se genera en la tarea 01 del proceso 05 (Data Collection) para el caso de uso 04 (vacunas)

Importante mantener los guiones bajos antes de proceso, tarea, archivo y nombre

### Proceso 05

In [21]:
caso <- "CU_18"
proceso <- '_05'
tarea <- "_18"
archivo <- ""
proper <- "_infraestructuras"
extension <- ".csv"

<font color='tomato'><b> OPCION A:</b> Uso del paquete "tcltk" para mayor comodidad</font>

* Buscar carpeta, escribir nombre de archivo SIN extensión (se especifica en el código)
* Especificar sufijo2 si es necesario
* Cambiar datos por datos_xx si es necesario

In [22]:
# file_save <- paste0(caso, proceso, tarea, tcltk::tkgetSaveFile(), proper, extension) 
# path_out <- paste0(oPath, file_save)
# write_csv(data_to_save, path_out)

# cat('File saved as: ')
# path_out

<font color='tomato'><b> OPCION B:</b> Especificar el nombre de archivo</font>

- Los ficheros de salida del proceso van siempre a Data/Output/.  

In [23]:
file_save <- paste0(caso, proceso, tarea, archivo, proper, extension) 
path_out <- paste0(oPath, file_save)
write_csv(data_to_save, path_out)

cat('File saved as: ')
path_out

File saved as: 

[1] "Data/Output/CU_18_05_18_infraestructuras.csv"

#### Copia del fichero a Input

Si el archivo se va a usar en otros notebooks, copiar a la carpeta Input (descomentar si hace falta)

In [24]:
# path_in <- paste0(iPath, file_save)
# file.copy(path_out, path_in, overwrite = TRUE)

## <font color=' #2874a6 '>Main Conclusions</font>

<font color='steelblue'>List and describe the general conclusions of the analysis carried out​.</font>

### Prerequisites

Para que funcione este código se necesita:

* Las rutas de archivos `Data/Input` y `Data/Output` deben existir (relativas a la ruta del _notebook_)
* El paquete tcltk instalado para seleccionar archivos interactivamente. No se necesita en producción.
* Los paquetes tcltk, sf, readr, dplyr, tidyr deben estar instalados.


### Configuration Management

This notebook has been tested with the following versions of R and packages. It cannot be assured that later versions work in the same way:
* R 4.2.2
* tcltk 4.2.2
* sf 1.0.9
* readr 2.1.3
* dplyr 1.0.10
* tidyr 1.3.0

### Data structures

#### Objeto `data`

- Hay 1138 filas con las variables:
	* grupo
	* tipo
	* nombre
	* CODMUN
	* DIRECCION
	* info
	* CMUN
	* CDIS
	* dist_aeropuerto
	* dist_intercambiador
	* dist_cercanias
	* dist_hospital
	* dist_universidad
	* dist_gob
	* dist_embajada
	* dist_ccomercial
	* X
	* Y


#### Observaciones generales sobre los datos

* Se ha reducido al número de puntos considerablemente


### Consideraciones para despliegue en piloto

* Ninguna

### Consideraciones para despliegue en producción

* Se deben crear los procesos ETL en producción necesarios para que los datos de entrada estén actualizados

## <font color=' #2874a6 '>Main Actions</font>

#### <font color=' #2874a6 '>Acciones done</font>

<font color='steelblue'>Indicate the actions that have been carried out in this process</font>

- Se han calculado los centroides de las estaciones de metro con más de una boca
- Se han calculad las distancias mínimas a puntos relevantes desde todas las bocas de metro

#### <font color=' #2874a6 '>Acctions to perform</font>

<font color='steelblue'>Indicate the actions that must be carried out in subsequent processes</font>

- Se deben obtener datos de demanda y capacidad diaria de todas las infraestructuras resultantes

## <font color='RED'>CODE TO DEPLOY (PILOT)</font>

A continuación se incluirá el código que deba ser llevado a despliegue para producción, dado que se entiende efectúa operaciones necesarias sobre los datos en la ejecución del prototipo

<font color=' red '><b>Description</b></font>

- No hay nada que desplegar en el piloto, ya que estos datos son estáticos o en todo caso cambian con muy poca frecuencia, altamente improbable durante el proyecto.


<font color=' red '><b>CODE</b></font>

In [28]:
# incluir código